bqplot/traits.py

```diff
@@ -152,49 +152,48 @@ def array_from_json(value, obj=None):
 def array_to_json(ar, obj=None, force_contiguous=True):
     if ar is None:
         return None
-    if ar.dtype.kind in ['O']:
-        has_strings = False
-        all_strings = True  # empty array we can interpret as an empty list
-        for el in ar:
-            if isinstance(el, six.string_types):
-                has_strings = True
-            else:
-                all_strings = False
-        if all_strings:
-            ar = ar.astype('U')
-        else:
-            if has_strings:
-                warnings.warn('Your array contains mixed strings and other types')
 
-    if ar.dtype.kind in ['S', 'U']:  # strings to as plain json
-        return ar.tolist()
-    type = None
+    array_type = None
 
     if ar.dtype.kind == 'O':
-        # If it's a Timestamp object
+        # Try to serialize the array of objects
+        isstring = np.vectorize(lambda x: isinstance(x, six.string_types))
```

Comment:
should we assume py3? If so, we can remove six?
```suggestion
        isstring = np.vectorize(lambda x: isinstance(x, str))
```

Reply:
Do you mind if we do that in another PR? Because it is kind-of unrelated to the change I am making.